permission/acl/acl.go

```diff
@@ -45,3 +46,25 @@ func ValidRawAccount(accountName string) error {
 	}
 	return nil
 }
+
+func validContractName(contractName string) error {
+	// param absence check
+	// contract naming rule check
+	contractSize := len(contractName)
+	contractMaxSize := utils.GetContractNameMaxSize()
+	contractMinSize := utils.GetContractNameMinSize()
+	if contractSize > contractMaxSize || contractSize < contractMinSize {
+		return fmt.Errorf("contract name length expect [%d~%d], actual: %d", contractMinSize, contractMaxSize, contractSize)
+	}
+	if m, _ := regexp.MatchString("^[a-z,A-Z,_]{1}[0-9,a-z,A-Z,_,.]{14}[0-9,a-z,A-Z,_]{1}$", contractName); !m {
```

Comment:
It would be better to use a pre-compiled regexp. 
see https://godoc.org/regexp#MustCompile

Reply:
> It would be better to use a pre-compiled regexp.
> see https://godoc.org/regexp#MustCompile

Thank you for you advice and I have fixed it.